.github/workflows/publish.yml

```diff
@@ -1,23 +1,52 @@
-name: Publish Docker Image
+name: Publish and Deploy Docker Image
 
 on:
   push:
     branches: [ "main" ]
 
+env:
+  AWS_REGION: ${{ secrets.AWS_REGION }}
+  AWS_ACCESS_KEY_ID: ${{ secrets.AWS_ACCESS_KEY_ID }}
+  AWS_SECRET_ACCESS_KEY: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
+
 jobs:
   publish:
     runs-on: ubuntu-latest
     steps:
-    - uses: actions/checkout@v3
-    - name: Set up Docker Buildx
+    -
+      uses: actions/checkout@v3
+    -
+      name: Set up Docker Buildx
       uses: docker/setup-buildx-action@v2
-    - name: Login to Public ECR
+    -
+      name: Login to Public ECR
       uses: docker/login-action@v2
       with:
         registry: public.ecr.aws
-        username: ${{ secrets.AWS_ACCESS_KEY_ID }}
-        password: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
+        username: ${{ env.AWS_ACCESS_KEY_ID }}
+        password: ${{ env.AWS_SECRET_ACCESS_KEY }}
       env:
         AWS_REGION: us-east-1
-    - name: Build and Publish
-      run: make publish-docker
+    -
+      name: Build and Publish
+      run: |
+        SHA_TAG=$(echo ${{ github.SHA }} | head -c 12)
+        RELEASE_TAG=$(cargo metadata --format-version=1 --no-deps | jq '.packages[0].version' | tr -d '"')
```

Comment:
This is not quite right. We need to only set the RELEASE_TAG if this is a release commit. Otherwise we will keep moving the release tag to the latest commit after the release was cut. 

Maybe a simple solution is to add an equivalent `Build and Publish` step to the `publish-release.yml` workflow that already exists? This way we only add a RELEASE_TAG when we actually have a release?

Reply:
FWIW  this is how the publish-release.yml knows its a release commit https://github.com/ceramicnetwork/rust-ceramic/blob/main/.github/workflows/publish-release.yml#L17

Reply:
Ah makes sense, ty!

Reply:
I like the idea of having a similar step in the release workflow.

Reply:
Actually, let me see if I can reuse the workflow.

Reply:
Ok, nvm, just seemed cleaner/easier to add the commit check you indicated to the current workflow.